### PR TITLE
[IOT Manager] Error log api time interval support

### PR DESCRIPTION
**Pull request by SagarKarwande**
- Add support to error logs api for time based querying
- fixes the pagination count for data and error log API's

### CODE REVIEW

File: apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex [hunk 1]
```diff
@@ -58,7 +58,15 @@ defmodule AcqdatCore.Model.IotManager.GatewayDataDump do
     count_query = logs_count_query(params)
 
     %Postgrex.Result{rows: [[total]]} = Ecto.Adapters.SQL.query!(Repo, count_query, [])
-    total_pages = floor(total / page_size) + rem(total, page_size)
+
+    extra_page =
+      if rem(total, page_size) > 0 do
```
Comment (BandanaPandey) [inline apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex:63]: @SagarKarwande if this if else condition is one-liner, we can also mimic it like ternary operator,
`extra_page = if  rem(total, page_size) > 0, do: 1, else: 0`
it looks more compact.

Thoughts ?

Comment (BandanaPandey) [reply apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex:63]: Also @SagarKarwande Can you explain here, why we had added this extra_page check ?

File: apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex [hunk 2]
```diff
@@ -110,6 +118,85 @@ defmodule AcqdatCore.Model.IotManager.GatewayDataDump do
...
+          inserted_at: inserted_at,
+          data: data,
+          error: error,
+          gateway_uuid: gateway_uuid
+        }
+      end)
+
+    count_query = error_logs_count_query(params)
+
+    %Postgrex.Result{rows: [[total]]} = Ecto.Adapters.SQL.query!(Repo, count_query, [])
```
Comment (BandanaPandey) [inline apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex:159]: @SagarKarwande this piece of code is redundant, we have used it above for latest logs, this page calculation as well as this return map.
Can you please make this code DRY ?

**Review by BandanaPandey — CHANGES_REQUESTED** [review]

File: apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex (hunk 1 above)
Comment (SagarKarwande) [reply apps/acqdat_core/lib/acqdat_core/iot-manager/model/gateway_data_dump.ex:63]: @BandanaPandey 
When the records cannot be divided exactly by the page size then we require extra page for the remaining records.
eg: If we have 22 records and page size of 10, then the last 2 records will come on 3rd page.

**Pull request merged**